File: squares-of-a-sorted-array/squares-of-a-sorted-array.cpp
class Solution {
public:
vector<int> sortedSquares(vector<int>& nums) {

    vector<int> numsList(nums.size());
    int low = 0;
    int high = nums.size()-1;
    int i = high;

    while(high >= low){
       if(abs(nums[high]) >= abs(nums[low])){
           numsList[i] = nums[high] * nums[high];
           high--;
       } else{
           numsList[i] = nums[low] * nums[low];
           low++;
       }
       i--;
    }
    return numsList;
}
};

File: squares-of-a-sorted-array/squares-of-a-sorted-array.rs
impl Solution {
    pub fn sorted_squares(nums: Vec<i32>) -> Vec<i32> {
        

    let mut squared_nums = Vec::with_capacity(nums.len());
        nums.into_iter().for_each(|num| squared_nums.push(num * num));
    
    squared_nums.sort();
    squared_nums
        
    }
}